### PR TITLE
Add Optional to config_file_mode and config_folder_mode params on windows

### LINKED ISSUE #155: Cannot install telegraf on Windows server because config_file_mode and config_folder_mode is undefined
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.18.0
- Puppet Bolt: 2.30.0
- Ruby: 2.5.0
- Distribution: Microsoft Windows Server 2019 Datacenter（GCE）
- Module version:3.1.0

## How to reproduce (e.g Puppet code you use)

```yaml
---
description: Install and configure telegraf

parameters:
  targets:
    type: TargetSpec
    description: The targets to configure

steps:
  - description: Install and configure telegraf
    name: configure
    targets: $targets
    resources:
      - class: telegraf

return: $configure
```

```powershell
bolt plan run telegraf -t localhost
```

## What are you seeing

- I cannot install telegraf
- It worked on windows as well after adding config_file_mode

## What behaviour did you expect instead

- I can install telegraf

## Output log

```
Failed on localhost:
  Apply failed to compile for localhost: Class[Telegraf]:
    parameter 'config_file_mode' expects a match for Stdlib::Filemode = Pattern[/\A(([0-7]{1,4})|(([ugoa]*([-+=]([rwxXst]*|[ugo]))+|[-+=][0-7]+)(,([ugoa]*([-+=]([rwxXst]*|[ugo]))+|[-+=][0-7]+))*))\z/], got Undef
    parameter 'config_folder_mode' expects a match for Stdlib::Filemode = Pattern[/\A(([0-7]{1,4})|(([ugoa]*([-+=]([rwxXst]*|[ugo]))+|[-+=][0-7]+)(,([ugoa]*([-+=]([rwxXst]*|[ugo]))+|[-+=][0-7]+))*))\z/], got Undef (line: 4, column: 5)
Failed on 1 target: localhost
```

### PR DESCRIPTION
**Pull request by jkkitakita**
#### Pull Request (PR) description

- Add Optional to `config_file_mode` and `config_folder_mode` params on windows

#### This Pull Request (PR) fixes the following issues

Fixes https://github.com/voxpupuli/puppet-telegraf/issues/155

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -320,7 +320,7 @@ The configuration generated with this module is only compatible with newer relea
 Please fork this repository, hack away on your branch, run the tests:
 
 ```shell
-$ bundle exec rake test acceptance
+$ bundle exec rake test spec/acceptance
```
Comment (bastelfreak) [inline README.md]:
```
bundle exec rake beaker
```
is probably the better way to handle this. Also our .github/CONTRIBUTUNG.md explains how to run the acceptance tests on different operating systems.

File: .puppet-lint.rc [hunk 2]
```diff
@@ -0,0 +1 @@
+--relative
```
Comment (bastelfreak) [inline .puppet-lint.rc]: can you explain why you changed this? It should not be required to run the tests.

Comment (jkkitakita) [reply .puppet-lint.rc]: When I tried to git commit, puppet-lint was executed in pre-commit and it failed.
Without this fix, I couldn't git commit, so I added it.
```
~/jkkitakita/puppet-telegraf master +1 ❯ git commit -m "hoge"
Running pre-commit hooks
Check for trailing whitespace....................[TrailingWhitespace] OK
Runs puppet-lint on modified files only..................[PuppetLint] FAILED
Errors on lines you didn't modify:

jkkitakita/puppet-telegraf/manifests/params.pp:5:7:ERROR: telegraf::params not in autoload module layout (autoloader_layout)

✗ One or more pre-commit hooks failed
```

File: README.md (hunk 1 above)
Comment (jkkitakita) [reply README.md]: Thank you for reviewing.
Then, this command was wrong in any case, so I fixed it.

**Comment by jkkitakita** [comment]
@kenyon @bastelfreak 
Please let me know if there are any implementations missing in this PR.

**Review by kenyon — APPROVED** [review]

**Comment by benh57** [comment]
Looks like this was approved, could someone merge?

**Pull request merged**